### PR TITLE
Make deployment of Opensearch optional

### LINKED ISSUE #1370: Make OpenSearch optional
**Is your feature request related to a problem? Please describe.**
This would give the option to better isolate which components to deploy and we do have a use for it. And then we wouldn't need to keep a separate branch where it is disabled.

**Describe the solution you'd like**
Make OpenSearch (and Fluentd in WC) optional.

**Definition of done:**
OpenSearch can be disabled via config.

### PR DESCRIPTION
**Pull request by Zash**
> [!warning]
> **This is public repository, ensure not to disclose:**
> - [ ] personal data beyond what is necessary for interacting with this pull request
> - [ ] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [x] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### Platform Administrator notice

Deploying Opensearch can now be disabled by setting `.opensearch.enabled=false` in `sc-config.yaml`

### What does this PR do / why do we need this PR?

This introduces a flag that determines whether Opensearch is to be deployed in the service cluster, allowing it to _not_ be.

- Fixes #1370

#### Screenshots

Not an actual screenshot

```
# yq4 -i '.opensearch.enabled=false' $CK8S_CONFIG_PATH/common-config.yaml

DELETED RELEASES:
NAME
opensearch-curator
opensearch-configurer
opensearch-dashboards
opensearch-securityadmin
opensearch-master
opensearch-secrets
```

#### Checklist

- [ ] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [x] The change is disruptive (if you flip the flag)
  - [x] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy

### CODE REVIEW

File: helmfile/values/grafana/grafana-dashboards.yaml.gotmpl [hunk 1]
```diff
@@ -27,6 +29,9 @@ disabledDashboards:
   - vulnerability-dashboard
   - thanos-sidecar-dashboard
   - thanos-bucket-replicate-dashboard
+{{- if .Values.opensearch.enabled }}
+  - opensearch-dashboard
+{{- end }}
```
Comment (aarnq) [inline helmfile/values/grafana/grafana-dashboards.yaml.gotmpl:34]:
```suggestion
  {{- if not .Values.opensearch.enabled }}
  - opensearch-dashboard
  {{- end }}
```

**Review by aarnq — COMMENTED** [review]
Should we move it to common-config and make wc fluentd shipping to opensearch conditional by that flag as well?
(Though since we still might want audit logging this may require a bit more work in the fluentd config. :thinking:)

File: helmfile/values/grafana/grafana-dashboards.yaml.gotmpl (hunk 1 above)
Comment (Zash) [reply helmfile/values/grafana/grafana-dashboards.yaml.gotmpl:34]: Boolean logic, every time!

File: helmfile/50-applications.yaml [hunk 2]
```diff
@@ -557,7 +561,7 @@ releases:
     app: fluentd
     component: fluentd-forwarder
   inherit: [ template: fluentd-elasticsearch ]
-  installed: {{ .Values.fluentd.enabled }}
+  installed: {{ and .Values.opensearch .Values.fluentd.enabled }}
```
Comment (linus-astrom) [inline helmfile/50-applications.yaml:564]: shoudn't this one be like the others?
```suggestion
  installed: {{ and .Values.opensearch.enabled .Values.fluentd.enabled }}
```

Comment (Zash) [reply helmfile/50-applications.yaml]: Thanks!

**Comment by OlleLarsson** [comment]
> ough since we still might want audit logging this may require a bit more work in the fluentd config.

I think we should make it so that if Opensearch is disabled, Fluentd should only output audit logs to s3.

**Comment by Zash** [comment]
Perhaps [these](https://github.com/elastisys/compliantkubernetes-apps/blob/cabf73e1afbbf74f84efee9a15a8d84cd387041a/helmfile/50-applications.yaml#L637-L660) should be conditional on `opensearch.enabled` as well?

**Comment by aarnq** [comment]
> Perhaps [these](https://github.com/elastisys/compliantkubernetes-apps/blob/cabf73e1afbbf74f84efee9a15a8d84cd387041a/helmfile/50-applications.yaml#L637-L660) should be conditional on `opensearch.enabled` as well?

No, not if audit logging is enabled, so it can still go under the main fluentd flag.

File: helmfile/values/fluentd/forwarder-workload-cluster.yaml.gotmpl [hunk 3]
```diff
@@ -90,18 +98,27 @@ extraConfigMaps:
...
     </match>
     {{- end }}
 
+      {{- if .Values.opensearch.enabled }}
     # Relabel others
     <match **>
       @type relabel
       @label @STORE
     </match>
+      {{- else }}
```
Comment (aarnq) [inline helmfile/values/fluentd/forwarder-workload-cluster.yaml.gotmpl:116]:
```suggestion
    {{- if .Values.opensearch.enabled }}
    # Relabel others
    <match **>
      @type relabel
      @label @STORE
    </match>
    {{- else }}
```

**Review by aarnq — APPROVED** [review]

**Review by Xartos — APPROVED** [review]

**Review by OlleLarsson — APPROVED** [review]

**Pull request merged**